controllers/databasecluster_controller.go

```diff
@@ -968,6 +968,30 @@ func (r *DatabaseClusterReconciler) genPXCBackupSpec(
 		}
 	}
 
+	if database.Spec.Backup.PITR.Enabled {
+		spec, backupStorage, err := r.genPXCStorageSpec(ctx, database.Spec.Backup.PITR.BackupStorageName, database.Namespace)
+		if err != nil {
+			return nil, errors.Join(err, errors.New("failed to get pitr storage"))
+		}
+
+		switch backupStorage.Spec.Type {
+		case everestv1alpha1.BackupStorageTypeS3:
+			spec.S3 = &pxcv1.BackupStorageS3Spec{
+				// use separate directory for binlogs
+				Bucket:            pitrBucketName(backupStorage.Spec.Bucket),
+				CredentialsSecret: backupStorage.Spec.CredentialsSecretName,
+				Region:            backupStorage.Spec.Region,
+				EndpointURL:       backupStorage.Spec.EndpointURL,
+			}
+		default:
+			return nil, fmt.Errorf("BackupStorage of type %s is not supported. PITR only works for s3 compatible storages", backupStorage.Spec.Type)
+		}
+
+		// create a separate storage for pxc pitr as the docs recommend
+		// https://docs.percona.com/percona-operator-for-mysql/pxc/backups-pitr.html
+		storages[pitrStorageName(backupStorage.Name)] = spec
```

Comment:
The BE is supposed to take care of purging the storage when pitr is disabled.
TBD in the scope of the BE ticket EVEREST-602

Reply:
UPD decision after sync with the Cloud team: no binlogs should be deleted.